File: SDKs/quotes.rest/3.1/go/pkg/models/operations/postquote.go
package operations

import (
	"openapi/pkg/models/shared"
)

type PostQuoteQueryParams struct {
	Author   *string `queryParam:"style=form,explode=true,name=author"`
	Language *string `queryParam:"style=form,explode=true,name=language"`
	Quote    string  `queryParam:"style=form,explode=true,name=quote"`
	Tags     *string `queryParam:"style=form,explode=true,name=tags"`
}

type PostQuoteSecurity struct {
	XTheySaidSoAPISecret shared.SchemeXTheySaidSoAPISecret `security:"scheme,type=apiKey,subtype=header"`
}

type PostQuoteRequest struct {
	QueryParams PostQuoteQueryParams
	Security    PostQuoteSecurity
}

type PostQuoteResponse struct {
	ContentType string
	StatusCode  int64
}

File: SDKs/quotes.rest/3.1/typescript/src/sdk/models/operations/postquote.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PostQuoteQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=author" })
  author?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=language" })
  language?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=quote" })
  quote: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=tags" })
  tags?: string;
}


export class PostQuoteSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  xTheySaidSoApiSecret: shared.SchemeXTheySaidSoApiSecret;
}


export class PostQuoteRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: PostQuoteQueryParams;

  @SpeakeasyMetadata()
  security: PostQuoteSecurity;
}


export class PostQuoteResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
